framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -360,7 +361,9 @@ static int getBoundIndexOffset(final List<? extends AnnotatedTypeMirror> upperBo
     static AnnotatedTypeMirror getTypeAtLocation(
             AnnotatedTypeMirror type, List<TypeAnnotationPosition.TypePathEntry> location) {
 
-        if (location.isEmpty()) {
+        if (location.isEmpty() && type.getKind() != TypeKind.DECLARED) {
+            // The annotation on a declared type might apply to the enclosing type, so that case is
```

Comment:
How about: `An annotation with an empty type path on a declared type applies to the outermost enclosing type. This logic is handled together with non-empty type paths in getLocationTypeADT. For other kinds of types no work is required for an empty type path.`